# Election Poll

Each day during 2019 an agency asked a hundred randomly selected people which party they would vote for if elections were held that day. Results of the poll were recorded in the following [file](data/electionpoll.csv). The Workers' Party asked for the report which they plan to use to improve their strategy for upcoming elections.

Fill in the missing values in the report for 2019:

**Solution:**

Import required libraries.

In [2]:
import pandas as pd

Import data.

In [3]:
filepath = "data/electionpoll.csv"
data = pd.read_csv(filepath)
data.head()

Date \ Party  National Party  People's Party  Alliance Party  Civic Party  \
0   2019-01-01               0              20              27           26   
1   2019-01-02               0              19              28           26   
2   2019-01-03               0              20              29           27   
3   2019-01-04               0              22              25           25   
4   2019-01-05               0              24              22           21   

   Conservative Party  Liberal Party  Workers' Party  
0                   1              3              23  
1                   1              4              22  
2                   0              3              21  
3                   1              2              25  
4                   4              1              28

In [12]:
data.describe()

National Party  People's Party  Alliance Party  Civic Party  \
count      365.000000      365.000000      365.000000   365.000000   
mean         1.789041        8.145205       19.638356    27.391781   
std          4.499315        8.240763        6.351054     3.901597   
min          0.000000        0.000000        2.000000    18.000000   
25%          0.000000        1.000000       17.000000    24.000000   
50%          0.000000        3.000000       21.000000    27.000000   
75%          0.000000       16.000000       24.000000    30.000000   
max         18.000000       24.000000       36.000000    37.000000   

       Conservative Party  Liberal Party  Workers' Party  
count          365.000000     365.000000      365.000000  
mean             2.263014      12.087671       28.684932  
std              1.648135       8.768261        3.790085  
min              0.000000       0.000000       14.000000  
25%              1.000000       3.000000       26.000000  
50%              2.000000      15.000000       29.000000  
75%              3.000000      20.000000       32.000000  
max              6.000000      29.000000       36.000000

In [9]:
data["Workers' Party"].mean()

28.684931506849313

- The arithmetic mean of votes for the Workers' Party is: 28.7 (rounded to one decimal place)

In [10]:
data["Workers' Party"].median()

29.0

- The median of votes for the Workers' party is: 29 (rounded to closest integer)

In [11]:
data["Workers' Party"].std()

3.7900850708390332

- The standard deviation of votes for the Workers' party is: 3.8 (rounded to one decimal place)

In [32]:
data['Date'] = pd.to_datetime(data['Date \ Party'])
march_data = data[data['Date'].dt.month == 3]
march_data.describe()

National Party  People's Party  Alliance Party  Civic Party  \
count       31.000000       31.000000       31.000000    31.000000   
mean         0.129032       18.580645       21.387097    27.225806   
std          0.499462        2.526845        2.538732     2.616737   
min          0.000000       14.000000       17.000000    22.000000   
25%          0.000000       16.500000       19.500000    25.500000   
50%          0.000000       19.000000       21.000000    28.000000   
75%          0.000000       21.000000       23.000000    29.000000   
max          2.000000       23.000000       26.000000    32.000000   

       Conservative Party  Liberal Party  Workers' Party  
count           31.000000      31.000000       31.000000  
mean             2.096774       1.935484       28.645161  
std              1.535249       1.504116        2.483927  
min              0.000000       0.000000       24.000000  
25%              1.000000       1.000000       27.000000  
50%              2.000000       2.000000       29.000000  
75%              3.000000       3.000000       30.500000  
max              5.000000       5.000000       33.000000

- The difference between the largest and the smallest number of votes for the Workers' party for March is: 9

In [13]:
data.max()

Date \ Party          2019-12-31
National Party                18
People's Party                24
Alliance Party                36
Civic Party                   37
Conservative Party             6
Liberal Party                 29
Workers' Party                36
dtype: object

- The largest number of votes that any party received on any day is: 37 votes.

In [18]:
data['Civic Party'].idxmax()

352

In [20]:
data["Date \ Party"][352]

'2019-12-19'

- That maximum was achieved on 2019-12-19 by Civic Party.

In [25]:
data.describe()

National Party  People's Party  Alliance Party  Civic Party  \
count      365.000000      365.000000      365.000000   365.000000   
mean         1.789041        8.145205       19.638356    27.391781   
std          4.499315        8.240763        6.351054     3.901597   
min          0.000000        0.000000        2.000000    18.000000   
25%          0.000000        1.000000       17.000000    24.000000   
50%          0.000000        3.000000       21.000000    27.000000   
75%          0.000000       16.000000       24.000000    30.000000   
max         18.000000       24.000000       36.000000    37.000000   

       Conservative Party  Liberal Party  Workers' Party  
count          365.000000     365.000000      365.000000  
mean             2.263014      12.087671       28.684932  
std              1.648135       8.768261        3.790085  
min              0.000000       0.000000       14.000000  
25%              1.000000       3.000000       26.000000  
50%              2.000000      15.000000       29.000000  
75%              3.000000      20.000000       32.000000  
max              6.000000      29.000000       36.000000

- The party with the largest difference between the maximum and minimum number of votes is Alliance Party. That difference is 34 votes.